src/lib.rs

```diff
@@ -277,88 +281,57 @@ enum IpfsEvent {
     Exit,
 }
 
-/// Configured Ipfs instace or value which can be only initialized.
-pub struct UninitializedIpfs<Types: IpfsTypes> {
-    repo: Repo<Types>,
-    span: Span,
-    keys: Keypair,
-    options: IpfsOptions,
-    repo_events: Receiver<RepoEvent>,
-}
-
-impl<Types: IpfsTypes> UninitializedIpfs<Types> {
+impl<Types: IpfsTypes> Ipfs<Types> {
     /// Configures a new UninitializedIpfs with from the given options and optionally a span.
     /// If the span is not given, it is defaulted to `tracing::trace_span!("ipfs")`.
     ///
     /// The span is attached to all operations called on the later created `Ipfs` along with all
     /// operations done in the background task as well as tasks spawned by the underlying
     /// `libp2p::Swarm`.
-    pub async fn new(options: IpfsOptions, span: Option<Span>) -> Self {
+    pub fn new(options: IpfsOptions, span: Option<Span>) -> Self {
```

Comment:
Working on the fetch_and_cat the second None looks quite confusing, and I have to look it up (probably because I didn't introduce it but :)). **Since** this line is being changed right now, I wish we could do something for this... Maybe add a method on IpfsOptions? `fn in_span(self, span: Span) -> IpfsOptions`?